ocrd_segment/repair.py

```diff
@@ -56,128 +62,75 @@ def process(self):
-                    region_mask = filters.maximum_filter(region_mask, (scale, 1), origin=0)
-                    region_mask = filters.minimum_filter(region_mask, (scale, 1), origin=0)
-                    region_mask = region_mask[scale:-scale, scale:-scale] # unprotect
-                    # find outer contour (parts):
-                    contours, _ = cv2.findContours(region_mask, cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_SIMPLE)
-                    # determine areas of parts:
-                    areas = [cv2.contourArea(contour) for contour in contours]
-                    total_area = sum(areas)
-                    if not total_area:
-                        # ignore if too small
-                        LOG.warning('Zero contour area in region "%s"', region.id)
-                        continue
-                    # pick contour and convert to absolute:
-                    region_polygon = None
-                    for i, contour in enumerate(contours):
-                        area = areas[i]
-                        if area / total_area < 0.1:
-                            LOG.warning('Ignoring contour %d too small (%d/%d) in region "%s"',
-                                        i, area, total_area, region.id)
-                            continue
-                        # simplify shape:
-                        polygon = cv2.approxPolyDP(contour, 2, False)[:, 0, ::] # already ordered x,y
-                        if len(polygon) < 4:
-                            LOG.warning('Ignoring contour %d less than 4 points in region "%s"',
-                                        i, region.id)
-                            continue
-                        if region_polygon is not None:
-                            LOG.error('Skipping region "%s" due to non-contiguous contours',
-                                      region.id)
-                            region_polygon = None
-                            break
-                        region_polygon = coordinates_for_segment(polygon, region_image, region_coords)
-                    if region_polygon is not None:
-                        LOG.info('Using new coordinates for region "%s"', region.id)
-                        region.get_Coords().points = points_from_polygon(region_polygon)
+                self.sanitize_page(page, page_id)
                 
             #
-            # plausibilize segmentation
+            # plausibilize region segmentation (remove redundant regions)
             #
-            mark_for_deletion = set()
-            mark_for_merging = set()
-
-            for i in range(0,len(regions)):
-                for j in range(i+1,len(regions)):
-                    LOG.info('Comparing regions "%s" and "%s"', regions[i].id, regions[j].id)
-                    region_poly1 = Polygon(polygon_from_points(regions[i].get_Coords().points))
-                    region_poly2 = Polygon(polygon_from_points(regions[j].get_Coords().points))
+            mark_for_deletion = list()
```

Comment:
I do not understand why you move from `set` to `list`.

Reply:
I removed the comment on that in the last commit: Because [generateDS does not make element classes hashable](https://github.com/OCR-D/core/issues/240).